### PR TITLE
global: community deletion and permissions

### PR DESCRIPTION
**Pull request by krzysztof**
* Adds deleting for communities.

* Adds simple permissions for owners and admin.

* Adds inclusion request mail notifications.

Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

**Comment by krzysztof** [comment]
Closes #23

Partially addresses #26 although the task for actual removal of communities in record metadata needs to be implemented

File: invenio_communities/config.py [hunk 1]
```diff
@@ -23,7 +23,12 @@
 
 import pkg_resources
 
-COMMUNITIES_REQUEST_EXPIRY_TIME = 7  # Default expiry time delta in days
+COMMUNITIES_REQUEST_EXPIRY_TIME = 7
+"""Time after which the inclusion requests automatically expire (in days)."""
+
+COMMUNITIES_DELETE_HOLDOUT_TIME = 7
+"""Time after which the communities marked for deletion are """
+"""hard-deleted from the system (in days)."""
```
Comment (jirikuncar) [inline invenio_communities/config.py]: `"""` works on multiple lines, hence starting new line with `"""` is not necessary.

File: invenio_communities/config.py [hunk 2]
```diff
@@ -23,7 +23,12 @@
 
 import pkg_resources
 
-COMMUNITIES_REQUEST_EXPIRY_TIME = 7  # Default expiry time delta in days
+COMMUNITIES_REQUEST_EXPIRY_TIME = 7
```
Comment (jirikuncar) [inline invenio_communities/config.py]: For consistency with other modules I would propose to use seconds to defining time.

File: invenio_communities/cli.py [hunk 3]
```diff
@@ -86,6 +87,7 @@ def request(community_id, record_id, accept):
         db.session.commit()
     else:
         InclusionRequest.create(community=c, record=record)
+        RecordIndexer().index_by_id(record.id)
```
Comment (lnielsen) [inline invenio_communities/cli.py]: After commit

File: invenio_communities/config.py (hunk 2 above)
Comment (krzysztof) [inline invenio_communities/config.py]: Using timedelta instead.

File: invenio_communities/models.py [hunk 4]
```diff
@@ -228,6 +261,12 @@ def add_record(self, record):
             record[communities_key] = record[communities_key] + [self.id, ]
         else:
             record[communities_key] = [self.id, ]
+        if current_app.config["COMMUNITIES_OAI_ENABLED"] and "_oai" in record:
+            oai_spec = get_oaiset_spec(self.id)
+            if "sets" in record["_oai"]:
+                record["_oai"]["sets"] = record["_oai"]["sets"] + [oai_spec, ]
```
Comment (krzysztof) [inline invenio_communities/models.py]: Can this be added like this?

File: invenio_communities/models.py [hunk 5]
```diff
@@ -228,6 +261,12 @@ def add_record(self, record):
             record[communities_key] = record[communities_key] + [self.id, ]
         else:
             record[communities_key] = [self.id, ]
+        if current_app.config["COMMUNITIES_OAI_ENABLED"] and "_oai" in record:
+            oai_spec = get_oaiset_spec(self.id)
```
Comment (jirikuncar) [inline invenio_communities/models.py]: inveniosoftware/invenio-oaiserver#62 has been merged so you can write something like:
```python
oaiset = OAISet.query.get(get_oaiset_spec(self.id))
if oaiset is None:
    # create one or log error
else:
    oaiset.add_record(record)
```

**Comment by krzysztof** [comment]
What is missing in this PR is the ckeditor...

**Pull request merged**